### PR TITLE
Stop local go binaries before overwriting. Fixes #1049.

### LINKED ISSUE #1049: Kill go binaries before upgrading them
Downloading Go pump binaries from: https://github.com/ecc1/medtronic/releases/download/v2018.05.24/medtronic-386-spi.tar.xz
./
./openaps.jq
./cgmpage
./sniff
./historypage
./pumphistory
./listen
./mmtune
./fakemeter
./cgmhistory
./mdt
Installing Go pump binaries ...
‘/root/go/bin/cgmhistory’ -> ‘/usr/local/bin/cgmhistory’
‘/root/go/bin/cgmpage’ -> ‘/usr/local/bin/cgmpage’
‘/root/go/bin/fakemeter’ -> ‘/usr/local/bin/fakemeter’
‘/root/go/bin/historypage’ -> ‘/usr/local/bin/historypage’
‘/root/go/bin/listen’ -> ‘/usr/local/bin/listen’
‘/root/go/bin/mdt’ -> ‘/usr/local/bin/mdt’
‘/root/go/bin/mmtune’ -> ‘/usr/local/bin/mmtune’
‘/root/go/bin/openaps.jq’ -> ‘/usr/local/bin/openaps.jq’
‘/root/go/bin/pumphistory’ -> ‘/usr/local/bin/pumphistory’
cp: cannot create regular file ‘/usr/local/bin/pumphistory’: Text file busy
‘/root/go/bin/sniff’ -> ‘/usr/local/bin/sniff’
Couldn't copy go/bin

### PR DESCRIPTION
**Pull request by larryprice**
If the local go binaries are running, the copy will fail and setup will die. This PR asks the user if it's okay to kill any running binaries and then kills them if requested.

My editor also auto-fixed some inconsistent spacing; lmk if this should be undone.

### CODE REVIEW

File: bin/oref0-setup.sh [hunk 1]
```diff
@@ -255,6 +255,32 @@ function add_to_crontab () {
     (crontab -l; crontab -l |grep -q "$1" || echo "$2" "$3") | crontab -
 }
 
+function request_stop_local_binary () {
+    if [[ -x /usr/local/bin/$1 ]]; then
+        if pgrep -x $1 > /dev/null; then
+            read -p "Need to stop $1 to complete installation. OK? [Y]N" -r
+            if [[ $REPLY =~ ^[^Nn]*$ ]]; then
```
Comment (scottleibrand) [inline bin/oref0-setup.sh]: Is this logic correct?  I'm not sure what the second ^ does, but I think we want this to be something like `if [[ $REPLY =~ ^[Yy]*$ ]]; then`, right?

File: bin/oref0-setup.sh [hunk 2]
```diff
@@ -255,6 +255,32 @@ function add_to_crontab () {
     (crontab -l; crontab -l |grep -q "$1" || echo "$2" "$3") | crontab -
 }
 
+function request_stop_local_binary () {
+    if [[ -x /usr/local/bin/$1 ]]; then
+        if pgrep -x $1 > /dev/null; then
+            read -p "Need to stop $1 to complete installation. OK? [Y]N" -r
```
Comment (alimhassam) [inline bin/oref0-setup.sh]: Would it be better to use the common prompt_yn functions?

File: bin/oref0-setup.sh (hunk 1 above)
Comment (larryprice) [reply bin/oref0-setup.sh]: The `^` tells it to check for "not 'N' or 'n'" - my thought was to have "yes" be the default so users could just hit enter. I'll explore the `prompt_yn` function, which looks like a better solution.

File: bin/oref0-setup.sh (hunk 2 above)
Comment (larryprice) [reply bin/oref0-setup.sh]: Agreed and updated!

**Pull request merged**